Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs

```diff
@@ -181,6 +188,46 @@ async void RunMBaaSAsync(object sender, EventArgs e)
         void SignOut(object sender, EventArgs e)
         {
             Auth.SignOut();
+            SignInInformationButton.Text = "User Not Authenticated";
+        }
+
+        private static byte[] Base64UrlDecode(string input)
+        {
+            var output = input;
+            output = output.Replace('-', '+'); // 62nd char of encoding
+            output = output.Replace('_', '/'); // 63rd char of encoding
+            switch (output.Length % 4) // Pad with trailing '='s
+            {
+                case 0: break; // No pad chars in this case
+                case 2: output += "=="; break; // Two pad chars
+                case 3: output += "="; break; // One pad char
+                default: throw new System.Exception("Illegal base64url string!");
+            }
+            var converted = Convert.FromBase64String(output); // Standard base64 decoder
+            return converted;
+        }
+
+        public static string Decode(string token, string key)
```

Comment:
Why doesn't `JwtSecurityTokenHandler` work here?

Reply:
Maybe [Jwt.Net](https://www.nuget.org/packages/JWT) or [Portable.JWT](https://www.nuget.org/packages/Portable.JWT) ?

Reply:
No please use the Microsoft one we use in ingestion if possible, https://www.nuget.org/packages/System.IdentityModel.Tokens.Jwt/

Reply:
MS one is incompatible. Requires .netstandart >= 1.4;
As for other dependencies - since the code for decoding is quite short, I decided not to add extra dependencies, and just code it in place.